pytorch_lightning/trainer/trainer.py

```diff
@@ -1457,7 +1457,7 @@ def _call_lightning_module_hook(
         pl_module = pl_module or self.lightning_module
 
         if pl_module is None:
-            raise MisconfigurationException("No lightning module is available to call hooks on")
+            raise Exception("No Lightning Module is available to call hooks on")
```

Comment:
TypeError would fit better here? cc @justusschock 

#10918

Reply:
I don't feel like TypeError exactly fits but I guess it is the best option